File: packages/ios/Sources/Snapfill/SnapfillWebView.swift
import Foundation
import WebKit

/// Convenience WKWebView subclass with built-in Snapfill support.
///
/// Usage:
/// ```swift
/// let webView = SnapfillWebView()
/// webView.snapfillDelegate = self
/// webView.load(URLRequest(url: URL(string: "https://example.com/checkout")!))
/// ```
public class SnapfillWebView: WKWebView {

    private lazy var snapfill: Snapfill = {
        let s = Snapfill(webView: self, options: snapfillOptions)
        return s
    }()

    public weak var snapfillDelegate: SnapfillDelegate? {
        get { snapfill.delegate }
        set { snapfill.delegate = newValue }
    }

    public var snapfillOptions: SnapfillOptions = SnapfillOptions()

    public override init(frame: CGRect = .zero, configuration: WKWebViewConfiguration = WKWebViewConfiguration()) {
        super.init(frame: frame, configuration: configuration)
        snapfill.attach()
    }

    @available(*, unavailable)
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    public func fillForm(_ mappings: [String: String]) {
        snapfill.fillForm(mappings)
    }

    public func reinject() {
        snapfill.reinject()
    }
}

File: packages/android/src/main/kotlin/com/snapfill/SnapfillWebView.kt
package com.snapfill

import android.content.Context
import android.util.AttributeSet
import android.webkit.WebView

/**
 * Convenience WebView subclass with built-in Snapfill support.
 *
 * Usage:
 * ```kotlin
 * val webView = SnapfillWebView(context)
 * webView.snapfillListener = object : SnapfillListener {
 *     override fun onFormDetected(fields: List<String>) { ... }
 * }
 * webView.loadUrl("https://example.com/checkout")
 * ```
 */
class SnapfillWebView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : WebView(context, attrs, defStyleAttr) {

    private val snapfill = Snapfill(this)

    var snapfillListener: SnapfillListener?
        get() = snapfill.listener
        set(value) { snapfill.listener = value }

    var snapfillOptions: SnapfillOptions = SnapfillOptions()

    init {
        snapfill.attach()
    }

    fun fillForm(mappings: Map<String, String>) {
        snapfill.fillForm(mappings)
    }

    fun reinject() {
        snapfill.reinject()
    }
}
